src/towncrier/_settings/load.py

```diff
@@ -147,13 +151,14 @@ def parse_toml(base_path: str, config: Mapping[str, Any]) -> Config:
     template = config.get("template", _template_fname)
     if template.startswith("towncrier:"):
         resource_name = "templates/" + template.split("towncrier:", 1)[1] + ".rst"
-        if not pkg_resources.resource_exists("towncrier", resource_name):
+        resource_path = resources.files("towncrier") / resource_name
+        if not resource_path.exists():
```

Comment:
Traversable objects are not guaranteed to have a `.exists()` and `.resolve()` method, eg if you're running towncrier from a .whl file

Reply:
see here for a porting guide for resource_filename: https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-filename

Reply:
I was vaguely aware but is that a regression and/or relevant in our use case?

Reply:
Particularly because we don't consume the file but only pass its path further. as_file makes no sense to me?

Reply:
yeah the previous resource_filename would extract this template from towncrier zips and wheels and place it in a temporary directory

Reply:
And then just leave it there??

Reply:
yeah it would just leave it there, the new api forces a context manager so the file gets cleaned up